src/ecalc/libraries/libecalc/common/libecalc/dto/components.py

```diff
@@ -467,14 +469,28 @@ def validate_unique_names(cls, values):
                 if isinstance(fuel_consumer, GeneratorSet):
                     for electricity_consumer in fuel_consumer.consumers:
                         names.append(electricity_consumer.name)
+                if fuel_consumer.fuel is not None:
+                    for fuel_type in fuel_consumer.fuel.values():
+                        # Need to verify that it is a different fuel
+                        if fuel_type is not None and fuel_type not in fuel_types:
```

Comment:
can fuel_type be None?

Reply:
why only add fuel_name if fuel_type not already added?

Reply:
A test failed due to fuel_type None, don´t remember what caused it.

Adding fuel_name only if it is a new fuel - it should be possible to re-use the same fuel for several consumers/installations. So, the duplicate name check is only done if it is a different/new fuel (e.g. some fuel parameters are different)